### ISSUE TITLE
System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access.

### ISSUE THREAD

**Issue by herebebeasties** [Cinchoo/ChoETL#75]
I'd like to try ChoETL but I'm failing at the first hurdle:
```
System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at ChoETL.ChoTypeDescriptor.Init(Type type)
   at ChoETL.ChoTypeDescriptor.GetProperties(Type type)
   at ChoETL.ChoJSONRecordConfiguration.DiscoverRecordFields(Type recordType, Boolean clear, List`1 recordFieldConfigurations)
   at ChoETL.ChoJSONRecordConfiguration.Init(Type recordType)
   at ChoETL.ChoJSONRecordConfiguration..ctor(Type recordType)
```
Looks like you have no thread-safety around all these static type caches. :-(

**Comment by Cinchoo** [comment]
can u pls post sample code along with package version u are using?

**Comment by Cinchoo** [comment]
Fix is applied to nuget package version (1.1.0.5-beta5), try and let me know.
